woocommerce/facebook-for-woocommerce
# Update token exchange endpoint

### Changes proposed in this Pull Request:

Must be tested together with https://github.com/Automattic/woocommerce-connect-server/pull/2563.

Refactors how oAuth access tokens are obtained. Instead of taking them from the incoming query string, a call to WooCommerce Connect Sever to exchange the incoming oAuth code for the access token is made.

### Detailed test instructions:

1. Set up an `api-staging` environment using the `woocommerce-connect-server`'s [sirt/facebook-access-token-exposure](https://github.com/Automattic/woocommerce-connect-server/pull/2563) branch.
2. Configure the extension to use api-staging WooCommerce Connect Server version by search-replacing api.woocommerce.com with api-staging.woocommerce.com through the includes/Handlers/Connection.php file of the plugin.
4. Try to connect your Facebook for WooCommerce plugin to Facebook.
5. Check your webserver's access log to see no access tokens exposed in the access log files.


### Additional details:


### Changelog entry

> Update - Access Token retrieval procedure.
